aptos-move/aptos-vm/src/move_vm_ext/session.rs

```diff
@@ -344,14 +392,27 @@ impl<'r, 'l> SessionExt<'r, 'l> {
             }
         }
 
-        for (addr, account_changeset) in resource_group_change_set.into_inner() {
-            let (_, resources) = account_changeset.into_inner();
-            for (struct_tag, blob_op) in resources {
-                let state_key =
-                    StateKey::access_path(ap_cache.get_resource_group_path(addr, struct_tag));
-                let op = woc.convert(&state_key, blob_op, false)?;
-                resource_write_set.insert(state_key, op);
-            }
+        match resource_group_change_set {
+            ResourceGroupChangeSet::V0(v0_changes) => {
+                for (state_key, blob_op) in v0_changes {
+                    let op = woc.convert(&state_key, blob_op, false)?;
+                    resource_write_set.insert(state_key, op);
+                }
+            },
+            ResourceGroupChangeSet::V1(v1_changes) => {
+                for (state_key, resources) in v1_changes {
+                    resource_group_write_set.insert(
+                        state_key.clone(),
+                        resources
+                            .into_resources()
+                            .into_iter()
+                            .map(|(struct_tag, blob_op)| {
+                                Ok((struct_tag, woc.convert(&state_key, blob_op, false)?))
```

Comment:
no need to convert each op - no distinction between creation / modification here & no deletion